core/dbt/parser/manifest.py

```diff
@@ -599,6 +651,115 @@ def write_manifest_for_partial_parse(self):
         except Exception:
             raise
 
+    def build_public_nodes(self) -> bool:
+        """This method loads the dependencies from dependencies.yml, reads in the
+        the publication artifacts and adds the PublicModels to the manifest
+        "public_nodes" dictionary. It returns a boolean that indicates that
+        public nodes have been rebuilt."""
+        public_nodes_changed = False
+
+        # Load the dependencies from the dependencies.yml file
+        dependencies_filepath = resolve_path_from_base(
+            DEPENDENCIES_FILE_NAME, self.root_project.project_root
+        )
```

Comment:
Not blocking: we may need to rework this to implement https://github.com/dbt-labs/dbt-core/issues/7484

Reply:
Yep. I do wonder if moving away from "get_full_manifest" to more directly constructing the ManifestLoader might be better if we're going to want to pass a lot of things in to get_full_manifest.